MaxText/max_utils.py

```diff
@@ -272,65 +272,36 @@ def initialize_jax_for_cpu():
-
-  local_id_file = epath.Path(raw_keys["local_checkpoint_directory"]) / ID_FILE
-  if local_id_file.exists():
-    max_logging.log("An ID file from a previous run exists, initializing JAX distributed runtime using the saved ID.")
-    process_id = int(local_id_file.read_text())
-    coordinator_address_file = _get_coordinator_address_file(raw_keys)
-    if process_id == 0:
-      coordinator_address = _get_coordinator_address_for_emergency_checkpointing(raw_keys)
-      coordinator_address_file.write_text(coordinator_address)
-    else:
-      coordinator_address = _retrieve_coordinator_address(coordinator_address_file)
-    max_logging.log(f"Using the saved process_id of {process_id} and the 0'th process's address {coordinator_address}"
-                    " to initialize JAX distributed runtime...")
-    jax.distributed.initialize(coordinator_address=coordinator_address, process_id=process_id)
+  process_id, coordinator_address = _retrieve_jax_init_info(raw_keys)
+
+  if process_id != "" and coordinator_address != "":
+    max_logging.log(f"Using {process_id} as the process_id and {coordinator_address} as the"
+                    " coordinator_address to initialize JAX distributed runtime...")
+    jax.distributed.initialize(coordinator_address=coordinator_address, process_id=int(process_id))
   else:
-    max_logging.log("No ID file from a previous run exists, initializing JAX distributed runtime without args.")
+    max_logging.log("Initializing JAX distributed runtime without args when emergency checkpointing is"
+                    " enabled. This should not happen and your workload may have unexpected behavior.")
     jax.distributed.initialize()
-    process_id = jax._src.distributed.global_state.process_id  # pylint: disable=protected-access
-    local_id_file.write_text(str(process_id))
 
   ocp.multihost.utils.initialize_runtime_to_distributed_ids()
 
 
-def _get_run_name(raw_keys):
-  if raw_keys["run_name"] != "":
-    return raw_keys["run_name"]
-  return os.environ.get("JOBSET_NAME")
-
-
-def _get_coordinator_address_file(raw_keys):
-  COORDINATOR_ADDRESS_FILE = "coordinator_address.txt"
-  return epath.Path(os.path.join(raw_keys["base_output_directory"], _get_run_name(raw_keys), COORDINATOR_ADDRESS_FILE))
-
-
-def _get_coordinator_address_for_emergency_checkpointing(raw_keys):
-  run_name = _get_run_name(raw_keys)
-  slice_id = os.environ.get('MEGASCALE_SLICE_ID')
-  worker_id = os.environ.get('TPU_WORKER_ID')
-  return f"{run_name}-slice-job-{slice_id}-{worker_id}.{run_name}:8476"
-
-
-def _retrieve_coordinator_address(coordinator_address_file):
-  for _ in range(30):
-    if coordinator_address_file.exists():
-      return coordinator_address_file.read_text()
+def _retrieve_jax_init_info(raw_keys):
+  """Retrieve JAX init info from a local file."""
+  JAX_INIT_INFO_FILE = "jax-init-info.txt"
+  local_jax_init_info_file = epath.Path(raw_keys["local_checkpoint_directory"]) / JAX_INIT_INFO_FILE
+  for i in range(900):
```

Comment:
I'm confused by this - isn't the file local? Why does it take time to find it? Can you add a comment here?

Reply:
Added the comment to explain.